### PR TITLE
Ensure the stock/sales are updated upon deleting an attendee from the admin

### PR DESCRIPTION
**Pull request by Camwyn**
🎫 https://central.tri.be/issues/102634#note-25

### CODE REVIEW

File: src/Tribe/RSVP.php [hunk 1]
```diff
@@ -2067,4 +2071,30 @@ protected function parse_ticket_quantity( $product_id ) {
...
+	 *
+	 * @param $attendee_id
+	 *
+	 * @return bool|void
+	 */
+	public function update_stock_from_attendees_page( $attendee_id ) {
+		$attendee = get_post( $attendee_id );
+
+		// Can't find the attendee post
+		if ( empty( $attendee ) || self::ATTENDEE_OBJECT !== $attendee->post_type ) {
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: Split this into two conditionals and `return false;`

File: src/Tribe/RSVP.php [hunk 2]
```diff
@@ -2067,4 +2071,30 @@ protected function parse_ticket_quantity( $product_id ) {
...
+			return;
+		}
+
+		$ticket_id = get_post_meta( $attendee->ID, '_tribe_rsvp_product', true );
+
+		// Orphan attendees? No event to update.
+		if ( empty( $ticket_id ) ) {
+			return;
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: `return false;`

File: src/Tribe/RSVP.php [hunk 3]
```diff
@@ -2067,4 +2071,30 @@ protected function parse_ticket_quantity( $product_id ) {
...
+		}
+
+		$this->update_sales_and_stock_by_order_status( $attendee->ID, 'no', $ticket_id );
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: `return` the result of this method.

File: src/Tribe/RSVP.php (hunk 1 above)
Comment (Camwyn) [reply src/Tribe/RSVP.php]: Why split it?

File: src/Tribe/RSVP.php [hunk 4]
```diff
@@ -2067,4 +2071,35 @@ protected function parse_ticket_quantity( $product_id ) {
...
+		if ( empty( $attendee ) ) {
+			return false;
+		}
+
+		// It's not an attendee post
+		if ( self::ATTENDEE_OBJECT !== $attendee->post_type ) {
+			return false;
+		}
+
+		$ticket_id = get_post_meta( $attendee->ID, '_tribe_rsvp_product', true );
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: use `self::ATTENDEE_PRODUCT_KEY` for `_tribe_rsvp_product`

**Pull request merged**